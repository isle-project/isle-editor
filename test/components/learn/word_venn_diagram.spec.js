// MODULES //

import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import LearnWordVennDiagram from '@isle-project/components/learn/word-venn-diagram/main.js';
import data from './../fixtures/tdm_sparse.json';


// TESTS //

describe( '<LearnWordVennDiagram />', function test() {
	it( 'renders a learning component', () => {
		const learning = renderer.create(
			<LearnWordVennDiagram
				tdm={data.tdm} vocabulary={data.vocabulary} nTexts={118}
			/>
		);
		expect( learning ).toMatchSnapshot();
	});
});
