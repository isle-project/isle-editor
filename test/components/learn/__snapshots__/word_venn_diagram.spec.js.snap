// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LearnWordVennDiagram /> renders a learning component 1`] = `
[
  <div
    className="card"
  >
    <h4
      className="card-header"
    >
      settings
    </h4>
    <div
      className="card-body"
    >
      <p>
        <span
          className="title"
        >
          number-of-words
        </span>
      </p>
      <div
        className="btn-group"
        role="group"
      >
        <button
          className="btn btn-success"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          one
        </button>
        <button
          className="btn btn-default"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          two
        </button>
        <button
          className="btn btn-default"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          three
        </button>
      </div>
      <div
        className="d-grid gap-1"
      >
        <div
          className="input text-container-div"
          style={{}}
        >
          <span>
            <label
              htmlFor="dynamic-text-input-1"
            >
              word 1
              :
            </label>
            <span />
          </span>
          <input
            className="text-input"
            id="dynamic-text-input-1"
            name="input"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            placeholder="enter-text"
            style={
              {
                "width": 120,
              }
            }
            type="text"
            value=""
          />
        </div>
        <div
          className="input"
          style={
            {
              "marginBottom": "4px",
              "marginTop": "4px",
            }
          }
        >
          <span>
            <label
              htmlFor="dynamic-number-input-1"
            >
              min-of-times:
            </label>
          </span>
          <span
            className="number-input-span"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <input
              autoComplete="off"
              className="number-number-input"
              disabled={false}
              id="dynamic-number-input-1"
              max={50}
              min={1}
              name="input"
              onBlur={[Function]}
              onChange={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              placeholder="0"
              step={1}
              style={
                {
                  "marginLeft": "24px",
                  "width": "80px",
                }
              }
              type="number"
              value={1}
            />
          </span>
        </div>
      </div>
      <button
        className="btn btn-primary"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        draw-venn-diagram
      </button>
    </div>
  </div>,
  <br />,
  <div
    className="card"
  >
    <h4
      className="card-header"
    >
      venn-diagram
    </h4>
    <div
      className="card-body"
      style={
        {
          "minHeight": 200,
        }
      }
    >
      <span
        style={
          {
            "position": "absolute",
            "right": 40,
            "top": 50,
          }
        }
      >
        total-texts
        : 
        118
      </span>
      <div
        id="dynamic-word-venn-diagram-1-venn"
      />
    </div>
  </div>,
]
`;
